python/equistore-operations/equistore/operations/_dispatch.py

```diff
@@ -101,12 +101,12 @@ def bincount(input, weights=None, minlength=0):
         raise TypeError(UNKNOWN_ARRAY_TYPE)
 
 
-def list_to_array(array, data: List):
+def list_to_array(array, data: List[List[int]]):
     """Create an object from data with the same type as ``array``."""
     if isinstance(array, TorchTensor):
-        return torch.Tensor(data, device=array.device, dtype=array.dtype)
+        return torch.tensor(data)
```

Comment:
```suggestion
        return torch.tensor(data).to(array.dtype).to(array.device)
```

If we want to stay true to the docstring and the previous implementation